### ISSUE TITLE
Warning: componentWillReceiveProps has been renamed

### ISSUE THREAD

**Issue by IftekherSunny** [JedWatson/react-select#3720]
Getting this warning !!!! 

**Warning: componentWillReceiveProps has been renamed**

[Screenshot](https://monosnap.com/direct/piO3bTBumGoDqwRktUH3St9Q90uRGa)

**Comment by eugenet8k** [comment]
```
react-dom.development.js:11494 Warning: componentWillReceiveProps has been renamed, and is not recommended for use. See https://fb.me/react-async-component-lifecycle-hooks for details.

* Move data fetching code or side effects to componentDidUpdate.
* If you're updating state whenever props change, refactor your code to use memoization techniques or move it to static getDerivedStateFromProps. Learn more at: https://fb.me/react-derived-state
* Rename componentWillReceiveProps to UNSAFE_componentWillReceiveProps to suppress this warning in non-strict mode. In React 17.x, only the UNSAFE_ name will work. To rename all deprecated lifecycles to their new names, you can run `npx react-codemod rename-unsafe-lifecycles` in your project source folder.

Please update the following components: Select
```

**Comment by jakeboone02** [comment]
This is fixed by #3487.

**Comment by NordlingDev** [comment]
@jakeboone02 - I'm on version 3.0.4 and it still warns about this. I can even see it being used in the source code?

**Comment by Oscarteg** [comment]
@NordlingDev That is because @jakeboone02 references a pullrequest but the pullrequest is not merged yet.

**Comment by rachelleahr** [comment]
Any idea when this will be merged? trying to remove these warnings...

**Comment by Oscarteg** [comment]
@rachelleahr Pull request #3716 fixes it and is merged, but I don't know when it will be released.

**Comment by absenteerealitycheck** [comment]
Any news on when this will be released?

**Comment by rachelleahr** [comment]
@gwyneplaine When is the next release planned?

**Comment by dbsimeonov** [comment]
Does anybody know what is the solution for it? Still having this warning

**Comment by qipoy** [comment]
it fixed? but I have installed react 16.9.0 and react-select 3.0.4 using NPM.  and I still got the warning


```
componentWillReceiveProps has been renamed, and is not recommended for use. See https://fb.me/react-async-component-lifecycle-hooks for details.

```

**Comment by Kiyozz** [comment]
Not fixed atm

**Comment by yanlee26** [comment]
1

**Comment by ranneyd** [comment]
+1

**Comment by ranneyd** [comment]
Everyone retweet this lmao

https://twitter.com/dustinpkane/status/1174838579056394241

**Comment by ranneyd** [comment]
So I'm going to drop this here.

https://www.npmjs.com/package/react-select-temp

I'm not happy about it either. I'm just using this to make the warning go away until the maintainers in this lib sort this out. I'm going to continue to monitor the threads and switch back to the real package once this one is ready.

I just took this PR, published to NPM, then updated this lib to use that one
https://github.com/JedWatson/react-input-autosize/pull/167

**Comment by razh** [comment]
We use https://github.com/ds300/patch-package to patch unsupported lifecycle methods.

**Comment by gwyneplaine** [comment]
Sorry for the delay everyone, we've now published 3.0.6 with @ranneyd's fix to react-input-autosize, thanks again :) 

Looking forward to refactoring rs to truly deprecate usage of these lifecycles soon, thank you al lfor your patience 🙏 it really is appreciated

**Comment by aleksfront** [comment]
Thanks for updates. I've just switched to 3.0.6 but still see same warning
![Screenshot 2019-10-01 at 11 37 07](https://user-images.githubusercontent.com/9607060/65947050-1867b380-e440-11e9-93bc-4869b7410cad.png)

**Comment by gwyneplaine** [comment]
@aleksfront  thanks for this, I've released 3.0.7 with a fresh yarn.lock which should result in resolved warnings, please let me know if the problem persists 👍

**Comment by kavindadilshan** [comment]
: componentWillReceiveProps has been renamed, and is not recommended for use. See https://fb.me/react-async-component-lifecycle-hooks for details.

* Move data fetching code or side effects to componentDidUpdate.
* If you're updating state whenever props change, refactor your code to use memoization techniques or move it to static getDerivedStateFromProps. Learn more at: https://fb.me/react-derived-state
* Rename componentWillReceiveProps to UNSAFE_componentWillReceiveProps to suppress this warning in non-strict mode. In React 17.x, only the UNSAFE_ name will work. To rename all deprecated lifecycles to their new names, you can run `npx react-codemod rename-unsafe-lifecycles` in your project source folder.

Please update the following components: SafeView


please give the answer

**Comment by ranneyd** [comment]
@kavindadilshan `SafeView` is not only no where in this project, but it's no where in its node modules. It's probably another component/library in your app.

@gwyneplaine everything works like a charm now. Thank you so much!

**Comment by mitchellhamilton** [comment]
Thanks for the issue, closing since this is now fixed!

**Comment by mreishus** [comment]
Looks like there are still some React Strict Mode warnings being generated as of the latest version today (3.0.8). There's an issue for them here: #3745

**Comment by An1mus** [comment]
+1

**Comment by vedees** [comment]
With the release of React 16.3, some new lifecycle methods have been introduced, and release of React 17 will deprecate some lifecycle method.

**getDerivedStateFromProps** is one of those newly introduced lifecycle method replacing **componentWillReceiveProps**, which has now become **UNSAFE_componentWillReceiveProps**.

Solution: rename `componentWillReceiveProps` to `UNSAFE_componentWillReceiveProps` or use getDerivedStateFromProps but this option will need refactoring your code. prefect solutuon: do not use classes

**Comment by ljharb** [comment]
Note that renaming to "UNSAFE_" is a breaking change since it drops support for < 16.3.

**Comment by JanaSundar** [comment]
Getting this warning in version 3.1.0

index.js:1 Warning: Using UNSAFE_componentWillReceiveProps in strict mode is not recommended and may indicate bugs in your code. See https://fb.me/react-unsafe-component-lifecycles for details.

* Move data fetching code or side effects to componentDidUpdate.
* If you're updating state whenever props change, refactor your code to use memoization techniques or move it to static getDerivedStateFromProps. Learn more at: https://fb.me/react-derived-state

Please update the following components: AutosizeInput, Select

**Comment by luanfelipecosta** [comment]
Why the issue is closed then? Im still getting error in 3.1.0.

**Comment by AbelFuentes** [comment]
3.1.0 and still with issues...

**Comment by razh** [comment]
- These warnings are appearing because you are running in `<React.StrictMode>`.
- [create-react-app v3.4.1](https://github.com/facebook/create-react-app/releases/tag/v3.4.1) enabled `<React.StrictMode>` by default. It was released about one month ago.
- This issue was closed because it was resolved for the case where strict mode is not enabled.
- There is currently a rewrite in progress to remove these deprecated lifecycle methods.

**Comment by bladey** [comment]
Hi all,

Thank you to everyone who has provided information above.

As you may be aware, this bug has been reported multiple times as per issue #4094. On behalf of the maintainers I apologise it's taken awhile to get to this.

A new master issue #4094 has since been created to help keep track of this bug going forward.

This new issue will exist as the source of truth going forward to investigate the issue, report findings, and implement a bug fix.

**We'll take into account all the details above while investigating.**

If you feel this issue has been wrongly closed as it isn't related to the new master issue #4094, please let us know and we'll take another look.

Again, thank you.

**Comment by tomsarduy** [comment]
Happy 2023... any news on this? 👀

**Comment by Methuselah96** [comment]
It should be fixed in the latest version of react-select.
